server/src/trainee/traineeController.ts

```diff
@@ -0,0 +1,187 @@
+import { getRandomString } from "../utils/stringUtils";
+import { createMailAccount, addMailAccountToMailingList } from "../utils/plesk";
+import { createMWUser } from "../utils/mediawiki";
+import { createNCUser } from "../utils/nextcloud";
+import * as traineeTypes from "./traineeTypes";
+import { PoolConnection } from "mysql";
+import * as authTypes from "./../global/auth/authTypes";
+import { canPermissionBeDelegated, doesPermissionsInclude } from "../utils/authUtils";
+
+/**
+ * Retrieves choices of mentor, internal project and department of all trainees of given generation
+ * @param req generation_ID
+ * @param res member ID, first and last name and choices
+ */
+export const retrieveTraineeChoice = (req: Request, res: Response): void => {
+  database
+    .query(
+      `SELECT mitgliedID, vorname, nachname,
+      wahl_mentor, wahl_mentor1, wahl_mentor2, wahl_mentor3,
+      wahl_internesprojekt, wahl_internesprojekt1, wahl_internesprojekt2, wahl_internesprojekt3,
+      wahl_ressort, wahl_ressort1, wahl_ressort2, wahl_ressort3
+      FROM  mitglied
+      INNER JOIN generation
+      ON mitglied.generation = generation.generationID
+      WHERE  generation.generationID = ?`,
+      [req.body.generation_ID]
+    )
+    .then((result: traineeTypes.GetTraineeChoiceResult) => {
+      res.status(200).json(result);
+    })
+    .catch((err) => {
+      res.status(500).send("Query Error");
+    });
+};
+
+/**
+ * Gets letter of motivation form trainees of given generation
+ * @param req generation_ID
+ * @param res member ID and 3 motivation texts
+ */
+export const retrieveTraineeMotivation = (req: Request, res: Response): void => {
+  database
+    .query(
+      `SELECT mitgliedID, wahl_internesprojekt1_motivation, wahl_internesprojekt2_motivation,
+      wahl_internesprojekt3_motivation
+      FROM mitglied
+      INNER JOIN generation
+      ON mitglied.generation = generation.generationID
+      WHERE generation.generationID = ?`,
+      [req.body.generation_ID]
+    )
+    .then((result: traineeTypes.GetTraineeMotivationResult) => {
+      res.status(200).json(result);
+    })
+    .catch((err) => {
+      res.status(500).send("Query Error");
+    });
+};
+
+/**
+ * Retrieves information about all generations
+ * @param req null
+ * @param res generationID, short name and deadlines
+ */
+export const retrieveGenerations = (req: Request, res: Response): void => {
+  database
+    .query(
+      `SELECT generationID, bezeichnung, bewerbung_start, bewerbung_ende, wwTermin,
+      auswahlWETermin, infoabendBesucher,
+      tuercode, wahl_start, wahl_ende
+      FROM generation WHERE 1`,
+      []
+    )
+    .then((result: traineeTypes.GetGenerations) => {
+      res.status(200).json(result);
+    })
+    .catch((err) => {
+      res.status(500).send("Query Error");
+    });
+};
+
+/**
+ * Sets "wahl_start" and "wahl_ende" for generation
+ * @param req wahl_start and wahl_ende in form of DATETIME, generationID
```

Comment:
If you want to describe what body attribute should be given for the request you can write it as follows:
```suggestion
 * @param {Request} req request object
 * @param {Date} req.body.votingStart
 * @param {Date} req.body.votingEnd
 * @param {number} req.body.generationID
```

Reply:
fixed